### PR TITLE
Fix insta elimination on re-registration bug

### PR DESCRIPTION
**Pull request by jbonilla-tao**
### Description
[Provide a brief description of the changes introduced by this pull request.]

### Related Issues (JIRA)
[Reference any related issues or tasks that this pull request addresses or closes.]

### Checklist
- [ ] I have tested my changes on testnet.
- [ ] I have updated any necessary documentation.
- [ ] I have added unit tests for my changes (if applicable).
- [ ] If there are breaking changes for validators, I have (or will) notify the community in Discord of the release.

### Reviewer Instructions
[Provide any specific instructions or areas you would like the reviewer to focus on.]

### Definition of Done
- [ ] Code has been reviewed.
- [ ] All checks and tests pass.
- [ ] Documentation is up to date.
- [ ] Approved by at least one reviewer.

### Checklist (for the reviewer)
- [ ] Code follows project conventions.
- [ ] Code is well-documented.
- [ ] Changes are necessary and align with the project's goals.
- [ ] No breaking changes introduced.

### Optional: Deploy Notes
[Any instructions or notes related to deployment, if applicable.]

/cc @mention_reviewer

### CODE REVIEW

File: vali_objects/utils/challengeperiod_manager.py [hunk 1]
```diff
@@ -81,6 +81,25 @@ def _prune_deregistered_metagraph(self, hotkeys=None):
...
+    def is_recently_re_registered(self, ledger, positions):
+        """
+        A miner can re-register and their perf ledger may still be old.
+        This function checks for that condition and blocks challeneg eperiod failure so that
+        the perf ledger can rebuild.
+        """
+        if ledger:
+            time_of_ledger_start = ledger.start_time_ms
+        else:
+            # No ledger? No edge case.
```
Comment (trdougherty) [inline vali_objects/utils/challengeperiod_manager.py:93]: lol

File: vali_objects/utils/challengeperiod_manager.py [hunk 2]
```diff
@@ -81,6 +81,25 @@ def _prune_deregistered_metagraph(self, hotkeys=None):
                 self.challengeperiod_success.pop(hotkey)
 
 
+    def is_recently_re_registered(self, ledger, positions):
```
Comment (trdougherty) [inline vali_objects/utils/challengeperiod_manager.py:84]: oh nice, yeah I think this looks good

**Review by trdougherty — APPROVED** [review]

File: vali_objects/utils/challengeperiod_manager.py [hunk 3]
```diff
@@ -81,6 +81,25 @@ def _prune_deregistered_metagraph(self, hotkeys=None):
...
+        """
+        A miner can re-register and their perf ledger may still be old.
+        This function checks for that condition and blocks challeneg eperiod failure so that
```
Comment (sli-tao) [inline vali_objects/utils/challengeperiod_manager.py:87]: nit: typo

**Comment by sli-tao** [comment]
it's breaking some of the challenge period tests

**Comment by jbonilla-tao** [comment]
> appears to break some of the challenge period tests where miners have a ledger but no positions.

fixed thanks

File: tests/vali_tests/test_challengeperiod_unit.py [hunk 4]
```diff
@@ -44,7 +46,7 @@ def setUp(self):
         self.DEFAULT_POSITION = Position(
             miner_hotkey="miner",
             position_uuid="miner",
-            orders=[],
+            orders=[Order(price=60000, processed_ms=self.START_TIME, order_uuid="initial_order", trade_pair=TradePair.BTCUSD, order_type=OrderType.LONG, leverage=0.1)],
```
Comment (trdougherty) [inline tests/vali_tests/test_challengeperiod_unit.py:49]: thanks for fixing these tests! 🤝

**Review by trdougherty — APPROVED** [review]

**Review by sli-tao — APPROVED** [review]

**Pull request merged**